### ISSUE TITLE
In GitVersion.MsBuild 5.6.3,  AfterTargets="GetVersion" causes msbuild warning '"WriteVersionInfoToBuildLog" task could not be loaded'

### ISSUE THREAD

**Issue by philipag** [GitTools/GitVersion#2533]
The following target dependency causes VS build to generate a warning: 

`<Target Name="CalculateVersionCodeProperties" AfterTargets="GetVersion;_CompileAppManifest" BeforeTargets="CoreCompile">`

The warning:

Warning		The "WriteVersionInfoToBuildLog" task could not be loaded from the assembly ...gitversion.msbuild\5.6.3\tools\net48/GitVersion.MsBuild.dll. Method 'RegisterTypes' in type 'GitVersion.MsBuild.GitVersionTaskModule' from assembly 'GitVersion.MsBuild, Version=5.6.3.0, Culture=neutral, PublicKeyToken=null' does not have an implementation. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.		

## Expected Behavior
No warning

**Comment by sjoerd222888** [comment]
I tested with all nuget package versions out and saw the same behaviour.

I assume GitVersion.MsBuild should replace GitVersionTask and has to be used the same way, even though I could not find any documentation. Is my assumption correct?

While  GitVersion.MsBuild does not yet work and is not documented, does it make sense to mark GitVersionTask deprecated? Because the replacement seems not ready to me.

**Comment by arturcic** [comment]
@sjoerd222888  `is not documented` I agree with that and there is an issue regarding updating the docs, but you need to understand we spend our free time on maintaining the project, and if we can get some help with that we highly apreciate that. Unfortunately we do not have the time to do everything on time so that's why we need the help of the community, that is how open source works.

`While GitVersion.MsBuild does not yet work ` I cannot agree with that, as out internal tests and integration tests all pass, as well as there are people that were able to do the switch, and are able to use the new package on environments the old one was not working at all.

**Comment by AshleighAdams** [comment]
It's worth noting that I experienced this bug when converting from `GitVersionTask` to `GitVersion.MsBuild` too, but restarting Visual Studio fixed the issue. The only other difference (perhaps regression?) I saw was a lot of `gitversion.json` files being produced.

**Comment by arturcic** [comment]
>  I saw was a lot of gitversion.json

It is on purpose because the implementation of the task is different, the only bug was where the file gets written to, but we addressed that

**Comment by sjoerd222888** [comment]
I was not able to get beyond the error message, even with cleaning. I will investigate whether I can somehow getting it to work. Just to make sure I am not in another case I post my error message:

```
Error	MSB4062	The "WriteVersionInfoToBuildLog" task could not be loaded from the assembly 
C:\Users\MyUserName\.nuget\packages\gitversion.msbuild\5.6.3\tools\net48/GitVersion.MsBuild.dll. 
Method 'RegisterTypes' in type 'GitVersion.MsBuild.GitVersionTaskModule' 
from assembly 'GitVersion.MsBuild, Version=5.6.3.0, Culture=neutral, PublicKeyToken=null' 
does not have an implementation. Confirm that the <UsingTask> declaration is correct,
that the assembly and all its dependencies are available, 
and that the task contains a public class that implements Microsoft.Build.Framework.ITask.

C:\Users\MyUserName\.nuget\packages\gitversion.msbuild\5.6.3\tools\GitVersion.MsBuild.targets	13	
```


@arturcic Thanks for your feedback, I understand it should work, at least basically. I see your tests work. My feedback was not appropriate with this respect.

**Comment by bddckr** [comment]
I think this issue needs more information to be able to resolve this: In my own solution I'm building net462, net5.0, monoandroid10, Xamarin Mac without any issues.

I've been getting issues like this before 5.6.3 as it looks like MSBuild's build processes can't load the same assembly for different target frameworks, once one was already loaded by that process. That was due to `GitVersionAssemblyFile` getting set to different dlls depending on the projects' target framework. That has since been resolved and is no longer an issue in 5.6.3 anymore.

For the whole solution it now just grabs one dll, i.e. **the target frameworks of your projects no longer matter**:
https://github.com/GitTools/GitVersion/blob/aaaed5ed55cd3fe58335635dbe4695554781652b/src/GitVersion.MsBuild/msbuild/tools/GitVersion.MsBuild.props#L19-L27

I'm on VS on Windows, using version 16.8.4. The error sounds like either the dll is broken, or perhaps you're on a preview VS release and there's a dependency issue? Perhaps try this to ensure you're trying from a clean state:
1. Turn off VS.
2. Clear your solution's build folders (all bin + obj folders).
3. Go to your NuGet packages folder (`%userprofile%/.nuget/packages` by default) and delete the `gitversion.msbuild` folder to ensure VS grabs a new copy.

**Comment by philipag** [comment]
@bddckr I tried to test whether this is a problem only in the VS pre-release.  So I rebuilt my solution with the VS release version and the problem was gone.  Then I rebuilt with the preview and the problem is still gone!

I don't know what caused the warning to go away. A windows update recently caused my computer to reboot so perhaps a reboot is helpful after uninstalling GitVersionTask and installing GitVersion.MsBuild 5.6.3?
